### PR TITLE
Documentation

### LINKED ISSUE #1: Create design documents for how the platform will operate
We should have some design documents for how the platform operates.
- flowcharts for flow of data and operations.
- infrastructure design
- services design

### PR DESCRIPTION
**Pull request by TroyKomodo**
This PR aims to provide some documents that people can look at to get a better understanding of what this project will do.

Closes #1, #3

### CODE REVIEW

**Comment by TroyKomodo** [comment]
If you can please can you go look at the documents I made, and try to see if you can understand how it works. If anything is unclear it means the documents arent good enough. So if you dont understand anything just say so and ill clarify further in the documents! @ScuffleTV/reviewers

**Review by 4w3Official — APPROVED** [review]
lgtm

**Comment by DuneWalkerz** [comment]
looks all good to me

**Comment by IS2511** [comment]
The general structure is pretty clear, but the nitty gritty details are missing for sure, and the only place they recide is still your head, so you'll probably have to write a lot more docs soon 😄

**Comment by TroyKomodo** [comment]
> The general structure is pretty clear, but the nitty gritty details are missing for sure, and the only place they recide is still your head, so you'll probably have to write a lot more docs soon 😄

Yeah that is true. I'll try write some more documents on some of the finer details.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# Snuffle Contribution Guide
```
Comment (mmattbtw) [inline CONTRIBUTING.md:1]:
```suggestion
# Scuffle Contribution Guide
```

**Review by mmattbtw — CHANGES_REQUESTED** [review]
typo

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+## Monorepo
+
+For starters you will notice that this project is a [monorepo](https://semaphoreci.com/blog/what-is-monorepo).
+
+This means that all our code for every project is stored here in this repo.
+
+We opted to use a monorepo for a few reasons:
+- We can minimize code duplication, since we can share everything between services and products.
+- We can easily test and integrate the entire platform.
+- We can easily maintain and contribute to multiple projects within a single PR or ticket.
```
[the remark this answers is not in the document]
Comment (TroyKomodo) [reply CONTRIBUTING.md:12]: Tickets are similar to issues, but tickets more so report that the issue has been acknowledged and is going to be worked on

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+There are also integration tests which will run on your PR to ensure that it passes all the tests.
+
+## Documentation
+
+When you make a PR you should also update the documentation for the subproject you are working on. This is not required during the development stage but it is encouraged since it makes it easier to review your PR. Once we get to the merging phase we would require you to update the documentation so it is ready to be merged.
+
+## Questions
+
+If you have any questions please ask in the discord server or create an issue on the repo or in the discussion section.
```
[the remark this answers is not in the document]
Comment (TroyKomodo) [reply CONTRIBUTING.md:74]: Good idea!

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+There should be one commit per ticket. If we need more tickets then we can create sub issues and tasks around those.
+
+So the relationship between tickets to commits is `many to one` where we can have many tickets in a single commit but only one commit per ticket. 
+
+You should try and break up the commits as one ticket per commit but sometimes the trivial tickets might be small enough where we can just combine them into a single commit. 
+
+A maintainer and or reviewer will advise you on what you should do to make your PR mergeable.
+
+However you do not need to do this for the development stage of your PR. While developing you can commit as many times as you want with any names you like, however once it is ready for merge someone will ask you to squash your commits into tickets and fix up the naming on them. Once that is done and CI passes we can then merge your contributions!
```
[the remark this answers is not in the document]
Comment (TroyKomodo) [reply CONTRIBUTING.md:56]: Well, any PR should be squashed. I'll link a guideline there.

File: CONTRIBUTING.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (sirinoks) [reply CONTRIBUTING.md:12]: So will it be a label on an issue? Like, how does a new person know what is a ticket and what is not?

**Review by sirinoks — APPROVED** [review]
Pog

**Review by mmattbtw — APPROVED** [review]

**Comment by IS2511** [comment]
`CONTRIBUTING.md` still missing: branching strategy/rules (and branch naming), versioning strategy (if any, maybe a problem for later).

**Review by 4w3Official — APPROVED** [review]
ApuBased

**Review by treuks — APPROVED** [review]

**Review by mmattbtw — APPROVED** [review]

**Comment by IS2511** [comment]
We could also move the `design` dir to a new `docs` dir but that could easily be done in the future if needed.

**Review by IS2511 — APPROVED** [review]

**Review by 4w3Official — COMMENTED** [review]
docs(design): Spelling errors as well as grammatical errors fixed

**Comment by TroyKomodo** [comment]
@4w3Official can u make a new PR with these changes and we can go from there.

**Comment by 4w3Official** [comment]
sure

**Pull request merged**